shell/platform/darwin/ios/framework/Source/FlutterViewController.mm

```diff
@@ -1630,6 +1631,26 @@ - (void)setupKeyboardAnimationVsyncClient {
   [_keyboardAnimationVSyncClient await];
 }
 
+- (void)keyboardAnimationUpdateViewportMetricsInFrameRasterizedCallback {
+  fml::closure platformTask = [weakSelf = [self getWeakPtr]] {
+    if (!weakSelf) {
+      return;
+    }
+    fml::scoped_nsobject<FlutterViewController> flutterViewController(
+        [(FlutterViewController*)weakSelf.get() retain]);
+    if (!flutterViewController) {
+      return;
+    }
+    [flutterViewController updateViewportMetrics];
+  };
+
+  flutter::Shell& shell = [_engine.get() shell];
+  fml::closure onFrameRasterizedCallback = [&shell, platformTask] {
+    shell.GetTaskRunners().GetPlatformTaskRunner()->PostTask(platformTask);
+  };
+  shell.AddOnFrameRasterizedCallback(onFrameRasterizedCallback);
```

Comment:
I think add the callback here might be a frame late? 
What I was thinking was the callback will be some sort of delegate method and runs every frame. The delegate then decide if the method body is a no-op (when there is no keyboard animation), or synchronously runs the keyboard animation on platform thread.